Makefile

```diff
@@ -1,26 +1,42 @@
+$(MAKE)FILES = $(shell find . -maxdepth 3 -type f -name Makefile)
+SUBDIRS   = $(filter-out ./,$(dir $($(MAKE)FILES)))
+ROOT_DIR = $(shell pwd)
+
 default: build
 
 all: build test
 
-test: build
-	cargo test
-	cargo test --features testutils
-
 build:
-	cargo build --target wasm32-unknown-unknown --release -p soroban-token-contract
-	cargo build --target wasm32-unknown-unknown --release -p soroban-cross-contract-a-contract
-	cargo build --target wasm32-unknown-unknown --release -p soroban-atomic-swap-contract
-	cargo build --target wasm32-unknown-unknown --release
-	cd target/wasm32-unknown-unknown/release/ && \
-		for i in *.wasm ; do \
-			ls -l "$$i"; \
-		done
+	@for dir in $(SUBDIRS) ; do \
+		cd $$dir; \
+		$(MAKE) build || break; \
+		cd $(ROOT_DIR); \
+	done
+
+test: build
+	@for dir in $(SUBDIRS) ; do \
+		cd $$dir; \
+		$(MAKE) test || break; \
+		cd $(ROOT_DIR); \
+	done
 
 watch:
-	cargo watch --clear --watch-when-idle --shell '$(MAKE)'
+	@for dir in $(SUBDIRS) ; do \
+		cd $$dir; \
+		$(MAKE) watch || break; \
+		cd $(ROOT_DIR); \
+	done
```

Comment:
Watch will block on the first call and won't call others, so I don't think this will work. We can drop the watch target, it is't often used as far as I know.